app/src/main/java/io/unityfoundation/dds/permissions/manager/model/applicationgrant/ApplicationGrantService.java

```diff
@@ -269,13 +269,17 @@ public HttpResponse deleteById(Long grantId) {
         } else {
             User user = securityUtil.getCurrentlyAuthenticatedUser().get();
 
-            if (!securityUtil.isCurrentUserAdmin() &&
-                    !groupUserService.isUserTopicAdminOfGroup(applicationGrantOptional.get().getPermissionsGroup().getId(), user.getId())
-            ) {
+            Long groupId = applicationGrantOptional.get().getPermissionsGroup().getId();
+
+            boolean isAuthorized = securityUtil.isCurrentUserAdmin() ||
+                    groupUserService.isUserTopicAdminOfGroup(groupId, user.getId()) ||
+                    groupUserService.isUserApplicationAdminOfGroup(groupId, user.getId());
```

Comment:
I believe the group check here is wrong.  The group should be the group for the application instead of the grant.

Reply:
Agreed. I'll update the code now.